app/upgrades/v410/upgrade.go

```diff
@@ -0,0 +1,46 @@
+package v400
+
+import (
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	"github.com/desmos-labs/desmos/v4/app/upgrades"
+)
+
+var (
+	_ upgrades.Upgrade = &Upgrade{}
+)
+
+// Upgrade represents the v4.1.0 upgrade
+type Upgrade struct {
+	mm           *module.Manager
+	configurator module.Configurator
+}
+
+// NewUpgrade returns a new Upgrade instance
+func NewUpgrade(mm *module.Manager, configurator module.Configurator) *Upgrade {
+	return &Upgrade{
+		mm:           mm,
+		configurator: configurator,
+	}
+}
+
+// Name implements upgrades.Upgrade
+func (u *Upgrade) Name() string {
+	return "v4.1.0"
+}
+
+// Handler implements upgrades.Upgrade
+func (u *Upgrade) Handler() upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		// Do nothing here as we don't have anything particular in this update
```

Comment:
Just confirm, when will `x/posts` upgrade to v2?

Reply:
The posts' module consensus version is already set at `2`. The update from 1 to 2 has been done inside the `v4.0.0` upgrade

Reply:
I thought we register 1 as consensus version [here](https://github.com/desmos-labs/desmos/blob/master/app/upgrades/v400/upgrade.go#L43), I thought registered module version is the same as consensus version. Are they different?

Reply:
Yes, they are. Consensus version is set within the module itself (as you can see [here](https://github.com/desmos-labs/desmos/blob/v4.0.1/x/posts/module.go#L33)) and is returned by the modules' `ConsensusVersion` method ([here](https://github.com/desmos-labs/desmos/blob/v4.0.1/x/posts/module.go#L175)). The line you have linked is used to tell the upgrade handler that before the upgrade it should consider the module to have consensus version `1`. This is to avoid running `InitGenesis` for that module. More on the topic can be read [here](https://github.com/cosmos/cosmos-sdk/blob/main/docs/core/upgrade.md#genesis-state)

Reply:
@RiccardoM Got it! Thanks for the explanation and the references.